src/lib.rs

```diff
@@ -247,6 +250,7 @@ impl Lzxd {
         // E8 fixups are disabled after 1GB of input data,
         // or if the chunk size is too small.
         if chunk_offset >= 0x4000_0000 || idata.len() <= 10 {
+            // NOTE: This should not be reached since our caller should be checking these conditions.
```

Comment:
Perhaps it'd make sense to panic instead then. What do you think?

Reply:
I was going to do that, but just decided against it since it seemed kind of pointless.
I figure it was better to either not panic or just remove this check outright and opted for the former.

Reply:
If it can happen but shouldn't, I'm inclined to say an assertion (panic) would be the right tool.

If this can't happen, then yes, we should just remove the code.

Would it be reasonable or unexpected for the user if we return `idata` here?

Reply:
Yeah I suppose we could just remove this.
I didn't want to add a panic because it's a runtime failure that could potentially be randomly uncovered by some user, say if the calling code were refactored and this became possible to hit.
However, this code shouldn't fail even if this precondition isn't met.